overlord/auth/auth.go

```diff
@@ -161,6 +165,62 @@ func newMacaroonAuthenticator(macaroon string, discharges []string) *MacaroonAut
 	}
 }
 
+// MacaroonSerialize returns a store-compatible serialized representation of the given macaroon
+func MacaroonSerialize(m *macaroon.Macaroon) (string, error) {
+	marshalled, err := m.MarshalBinary()
+	if err != nil {
+		return "", err
+	}
+	encoded := base64.URLEncoding.EncodeToString(marshalled)
+	return encoded, nil
+}
+
+// base64Decode decodes base64 data that might be missing trailing pad characters
+// (copied from macaroon package; store serialized macaroons miss trailing padding)
+func base64Decode(b64String string) ([]byte, error) {
```

Comment:
Will that handle base64 data missing trailing pad characters? The macaroon library we use server-side is serializing the macaroons data in that situation. This specific helper handles this as expected (while using base64.URLEncoding.Decode was failing).